enhancements/kueue/openshift-kueue.md

```diff
@@ -0,0 +1,685 @@
+  - haircommander
+  - rphillips
+  - varshaprasad96
+approvers: # A single approver is preferred, the role of the approver is to raise important questions, help ensure the enhancement receives reviews from all applicable areas/SMEs, and determine when consensus is achieved such that the EP can move forward to implementation.  Having multiple approvers makes it difficult to determine who is responsible for the actual approval.
+  - mrunalp
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - joelspeed
+creation-date: 2025-02-19
+last-updated: 2025-02-19
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/OCPSTRAT-1641
+see-also:
+  - "https://github.com/openshift/enhancements/pull/1736"
+replaces:
+  - "NA"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# Bringing Kueue into Openshift
+
+## Summary
+
+We would like to bring Kueue into the core platform as a managed operator.
+Kueue will be hosted as a Red Hat ecosystem operator in OperatorHub.
+
+## Motivation
+
+Kueue has wide applications across various projects in the openshift ecosystem. 
+We want to bring them as a core platform so we can better support them in various areas. 
+Openshift AI (RHOAI) uses kueue for AI workloads but we are also seeing autoscaling, multicluster and general batch uses. 
+To better support various teams in their exploration or productization of Kueue, it is important to bring Kueue into the core platform.
+
+### High Level View of Kueue
+
+Kueue's [website](https://kueue.sigs.k8s.io/docs/overview/) provides a list of their functionality.
+
+Kueue provides the following functionality:
+
+- Job Management
+- Advanced Resource Management
+- Integrations with common workloads
+- Advanced Autoscaling support
+- All-or-nothing with Ready Pods
+- Partial Admission and dynamic reclaim
+- Mixing Training and Inferencing
+- Multi-cluster job dispatching
+- Topology-Aware Scheduling
+
+Kueue uses [Resource Flavors](https://kueue.sigs.k8s.io/docs/concepts/resource_flavor/) to describe what resources are available on a cluster to better support heterogenous clusters. 
+Resource flavors are cluster scoped and set up by the cluster adminstrator.
+
+Kueue uses [Cluster Queues](https://kueue.sigs.k8s.io/docs/concepts/cluster_queue/) to governs a pool of resources, defining usage limits and fair sharing rules.
+Cluster Queues are cluster scoped. Cluster queues are set up by a cluster administrator.
+
+Kueue uses [Local Queues](https://kueue.sigs.k8s.io/docs/concepts/local_queue/) to group closely related workloads belonging to a single tenant.
```

Comment:
Probably an implementation detail - In OCP terms (as well as in upstream) - Local Queue can be created either by cluster admin or the namespace admin.
Whereas in RHOAI - currently, only the cluster admin can create this resource. We aggregate the [local queue_editor](rbac.kueue.x-k8s.io/batch-admin: "true") role with `rbac.kueue.x-k8s.io/batch-admin` annotation to only cluster admins. 

In RHOAI namespaces (or projects) can be created by a normal user too. The persons on who can create what resources and relevant RBAC required for that also needs to be clarified.

Reply:
I added a section on this. https://github.com/openshift/enhancements/pull/1759/commits/89d5105bf9c58f0f1d8057ee5774ca819d4f0dc5